code/game/objects/items/airbubble.dm

```diff
@@ -0,0 +1,462 @@
+// Regular airbubble - folded
+/obj/item/airbubble
+	name = "air bubble"
+	desc = "Special air bubble designed to protect people inside of it from decompressed enviroments."
+	icon = 'icons/obj/airbubble.dmi'
+	icon_state = "airbubble_fact_folded"
+	w_class = ITEMSIZE_NORMAL
+	var/used = FALSE
+	var/ripped = FALSE
+	var/obj/item/weapon/tank/internal_tank
+
+/obj/item/airbubble/Destroy()
+	qdel(internal_tank)
+	return ..()
+
+/obj/item/airbubble/attack_self(mob/user)
+	var/obj/structure/closet/airbubble/R = new /obj/structure/closet/airbubble(user.loc)
+	if(!used && !ripped)
+		internal_tank = new /obj/item/weapon/tank/emergency_oxygen/double(src)
+	R.internal_tank = internal_tank
+	internal_tank.forceMove(R)
+	internal_tank = null
+	R.add_fingerprint(user)
+	R.ripped = ripped
+	R.update_icon()
+	qdel(src)
```

Comment:
Shouldn't you also do a `user.drop_from_inventory(src)`? Otherwise a ghost object may persist.